### PR TITLE
Update to LLVM 18 (and Rust 1.78)

### LINKED ISSUE #290: Failing test: felt252_to_bool_bug
Failing program:

```
use array::ArrayTrait;

        fn felt_to_bool(x: felt252) -> bool {
            x == 1
        }

        fn run_test(a: felt252) -> bool {
            felt_to_bool(a)
        }
```
with input `1` when comparing to cairo-vm output

Test can be found in `tests/boolean.rs felt252_to_bool_bug` 

comparing a felt252 == 1 will lead to wrong results

### PR DESCRIPTION
**Pull request by edg-l**
Fixes #290

### CODE REVIEW

**Comment by edg-l** [comment]
Status: when having a felt as second argument it seems to not get passed correctly, this is probs an issue in argumentmapper

**Comment by edg-l** [comment]
Should be ready

**Review by igaray — DISMISSED** [review]

File: src/libfuncs/felt252.rs [hunk 1]
```diff
@@ -513,47 +520,19 @@ pub mod test {
...
-            &FELT252_ADD,
-            "run_test",
-            &[JitValue::felt_str("0"), JitValue::felt_str("0")],
-            JitValue::felt_str("0"),
-        );
-    }
-
     #[test]
     fn felt252_add() {
+        // run_program_assert_output(
```
Comment (juanbono) [inline src/libfuncs/felt252.rs:525]: why this is commented?

**Review by juanbono — APPROVED** [review]

**Review by igaray — APPROVED** [review]

File: src/starknet.rs [hunk 2]
```diff
@@ -15,14 +15,14 @@ pub struct ArrayAbi<T> {
 
 /// Binary representation of a `Felt` (in MLIR).
 #[derive(Debug, Clone)]
-#[cfg_attr(target_arch = "x86_64", repr(C, align(8)))]
+#[cfg_attr(target_arch = "x86_64", repr(C, align(16)))]
 #[cfg_attr(not(target_arch = "x86_64"), repr(C, align(16)))]
```
Comment (azteca1998) [inline src/starknet.rs:19]: This is no longer necessary (the `cfg_attr`).

File: src/starknet.rs [hunk 3]
```diff
@@ -15,14 +15,14 @@ pub struct ArrayAbi<T> {
...
 pub struct Felt252Abi(pub [u8; 32]);
 /// Binary representation of a `u256` (in MLIR).
 // TODO: This shouldn't need to be public.
 #[derive(Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
 #[cfg_attr(feature = "with-serde", derive(serde::Serialize, serde::Deserialize))]
-#[cfg_attr(target_arch = "x86_64", repr(C, align(8)))]
+#[cfg_attr(target_arch = "x86_64", repr(C, align(16)))]
 #[cfg_attr(not(target_arch = "x86_64"), repr(C, align(16)))]
```
Comment (azteca1998) [inline src/starknet.rs:26]: This is no longer necessary (the `cfg_attr`).

**Review by azteca1998 — APPROVED** [review]

File: src/executor/aot.rs [hunk 4]
```diff
@@ -247,7 +247,7 @@ mod tests {
             tag: 0,
             value: JitValue::Struct {
                 fields: vec![JitValue::Felt252(
-                    syscall_handler.get_block_hash(0, &mut 0).unwrap(),
+                    syscall_handler.get_block_hash(1, &mut 0).unwrap(),
```
[the remark this answers is not in the document]
Comment (edg-l) [reply src/executor/aot.rs:250]: i changed the testsyscall handler to return its input, so we can test with more values

**Pull request merged**